extmod/network_lwip.c

```diff
@@ -90,6 +100,241 @@ mp_obj_t mod_network_nic_ifconfig(struct netif *netif, size_t n_args, const mp_o
+                            MP_OBJ_NEW_SMALL_INT(netif_ip6_addr_pref_life(netif, i)), // preferred
+                            MP_OBJ_NEW_SMALL_INT(netif_ip6_addr_valid_life(netif, i))
+                        };
+                        addrs[n_addrs++] = mp_obj_new_tuple(4, tuple);
+                    }
+                }
+                return mp_obj_new_list(n_addrs, addrs);
+            }
+            #endif
+            case MP_QSTR_gw4: {
+                return netutils_format_ipv4_addr((uint8_t *)&netif->gw, NETUTILS_BIG);
+            }
+            default: {
+                mp_raise_ValueError(MP_ERROR_TEXT("unexpected key"));
+                break;
+            }
+        }
+        return mp_const_none;
+    } else {
+        // Set config value(s)
+        if (n_args != 0) {
+            mp_raise_TypeError(MP_ERROR_TEXT("can't specify pos and kw args"));
+        }
+
+        for (size_t i = 0; i < kwargs->alloc; ++i) {
+            if (MP_MAP_SLOT_IS_FILLED(kwargs, i)) {
+                mp_map_elem_t *e = &kwargs->table[i];
+                switch (mp_obj_str_get_qstr(e->key)) {
+                    case MP_QSTR_dhcp4: {
+                        if (e->value == mp_const_true) {
+                            if (dhcp_supplied_address(netif)) {
+                                dhcp_renew(netif);
+                            } else {
+                                dhcp_release_and_stop(netif);
+                                dhcp_start(netif);
+                            }
+                            uint32_t start = mp_hal_ticks_ms();
+                            while (!dhcp_supplied_address(netif)) {
+                                if (mp_hal_ticks_ms() - start > 10000) {
+                                    mp_raise_msg(&mp_type_OSError, MP_ERROR_TEXT("timeout waiting for DHCP to get IP address"));
+                                }
+                                mp_hal_delay_ms(100);
+                            }
+                        } else {
+                            dhcp_release_and_stop(netif);
+                        }
+                        break;
+                    }
+                    #if LWIP_IPV6_DHCP6
+                    case MP_QSTR_dhcp6: {
+                        dhcp6_disable(netif);
+                        dhcp6_enable_stateless(netif);
+                        break;
+                    }
+                    #endif
+                    #if LWIP_IPV6_AUTOCONFIG
+                    case MP_QSTR_autoconf6: {
+                        netif_set_ip6_autoconfig_enabled(netif, e->value == mp_const_true);
+                        if (e->value == mp_const_true) {
+                            nd6_restart_netif(netif);
+                            uint32_t start = mp_hal_ticks_ms();
+                            while (1) {
+                                int found = 0;
+                                for (i = 1; i < LWIP_IPV6_NUM_ADDRESSES; i++) {
```

Comment:
Please add a comment for this loop... I think it's "See if we now have a valid non-static address on this interface".

Reply:
Also I see the comment later explains why this loop starts at 1, might be worth duplicating here (wasn't obvious to me).